lib/super_digit.rb

```diff
@@ -1,15 +1,21 @@
 # Superdigit
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(n)
+# Space Complexity - O(n)
 def super_digit(n)
-   
+  return n if n < 10
+
+  sum = n.to_s.split('').map { |num| num.to_i }.sum
+
+  return super_digit(sum)
 end
   
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(n)
+# Space Complexity - O(n)
 def refined_super_digit(n, k)
```

Comment:
This works, but it's O( (dk)^2) for time complexity, you can do better O(d^2 + k^2) with dynamic programming.